Articles/Defining and Using Observables.md

```diff
@@ -0,0 +1,341 @@
+### Definition Example: Observable by Object
+Defining an observable by object is in a metaschema object definition file. The definition is done by adding an `observable` field directly in the object definition object. This also works for object definitions that extend another, including the special patch case. For both regular and patch extends cases, the observable definition adds or replaces any existing object-level observable definition.
+
+This, along with defining observables by dictionary type, are the broadest ways to define observables. All instances attributes of this type (regardless of attribute name) become observables.
+
+Example in object definition file `objects/container.json`:
+```jsonc
+{
+  "name": "container",
+
+  // This is the observable definition for this object
+  "observable": 27,
+
+  // ... other object definition fields
+}
+```
+
+### Definition Example: Observable by Event Class Attribute / Observable by Object Attribute
+Observables can be defined in event class or object attributes. These are very similar and so are described together here. Defining observables this way limits the scope where these observables will occur. Note that definitions of this type are for attributes directly defined in the event class or object, not for attributes in a nested structure.
+
+Defining observables in event class or object attributes works just as with any other attribute definition at these levels: the fields defined override the same field defined in the dictionary or any event classes / objects the current item is derived from.
+
+Definitions of this type work for event class and object `extends` definitions both for the normal case (subclass / subtype) as well as the "patch extends" case, however "hidden" event classes and objects are not supported due to the potential of colliding observable `type_id` values. See [Appendix 2: Hidden Types](#appendix-2-hidden-types) for more about this issue.
+
+Example in object definition file `objects/cve.json`:
+```jsonc
+{
+  "name": "cve",
+  // ... other object definition fields
+  "attributes": {
+    // ... other attributes
+    "uid": {
+      "caption": "CVE ID",
+      // ... other attribute fields
+
+      // This defines uid as an observable when directly inside a cve object
+      "observable": 18
+    },
+    // ...
+  },
+  // ...
+}
+```
+
+Event classes work identically: attribute observable definitions occur in attribute detail objects inside the `attributes` object.
+
+### Definition Example: Observable by Class-Specific Attribute Path
+This last observable definition type allows defining an observable for an attribute inside a nested structure for an event class, though it can be used for attributes directly defined in the class (in this case being a alternative to defining event class attribute observables). This type of definition is done by adding an `observables` object to the event class definition that maps from attribute paths to observable `type_id` values.
```

Comment:
Same as above: "...adding an `observables` object to the event class definition..." - mixing up JSON objects with previous use of OCSF objects (which of course are JSON objects too, but it's confusing).  If I'm understanding this one.

Reply:
Reworded and changed "object" to "JSON object". Hopefully this reads better.

> This type of definition is done by adding an top-level `observables` field to the event class definition whose value is a JSON object that maps from attribute paths to observable `type_id` values.